### ISSUE TITLE
Rotate & perspective: auto-structure preserves prior adjustments; manual does not

### ISSUE THREAD

**Issue by n8willis** [darktable-org/darktable#12373]
This is on 4.0.0 on Ubuntu 20.04.

I have been using rotate & perspective as part of a LONG workflow to clean up a big set of document images for eventual OCR.

That involves perspective correction to square-up the corners.

For images where I use the "automatically determine structure" tool, applying the calculated perspective correction operates as expected.

For images where I use the "manual" structure tools, applying the calculated perspective correction makes the previous adjustment -- level correction -- rollback/disappear (I'm not being too picky on the terminology on that last part; it's still visible in the history stack, but it isn't being applied).

**Screenshots**
![darktable-persp](https://user-images.githubusercontent.com/2192132/186131942-8e7e0cd7-acf7-40f3-8118-8be2671a5a58.png)

Example in this screenshot. Left-hand image is one where I drew the perspective-structure manually via the rectangle tool. Right-hand image is one where I used the "auto" tool.

Prior to applying the correction, both images had color-curves matching the one on the right. Applying it reverted the left image to what you see.

No other adjustments made to either image.

**Comment by n8willis** [comment]
Still an issue.

**Comment by n8willis** [comment]
> Still an issue.

^

**Comment by Nilvus** [comment]
@n8willis: as 4.2 had been released on wednesday, could you confirm that you tested that issue on 4.2 release?

**Comment by n8willis** [comment]
> @n8willis: as 4.2 had been released on wednesday, could you confirm that you tested that issue on 4.2 release?

@Nilvus can you summarize what relevant bits changed in 4.2 in this regard? It's not clear to me whether this is more likely to be a UI problem or a core-processing-engine issue, so I'd like to understand what you feel is more likely in light of the 4.2 release cycle. Thanks for your help.

**Comment by Nilvus** [comment]
Sorry, I don't have the list, it's just good to test again after some months and new release.

**Comment by n8willis** [comment]
For the record, this is still broken.
